assets/prototype/domain/eventEditor/ticketsList/ticketPriceCalculator/TicketPriceCalculatorForm.js

```diff
@@ -1,201 +1,66 @@
-		desc: '',
-		isBasePrice: false,
-		isDeleted: false,
-		isDefault: false,
-		isDiscount: false,
-		isPercent: false,
-		name: '',
-		priceType: 5,
-		order: 999,
-		wpUser: 1,
-	};
-	const renderPriceRow = (price, btn = 'trash') => {
-		const options = price.priceType === 1 ? allOptions : modifierOptions;
-		const sign = price.isPercent ? percentSign : currencySign;
-		let b4Price = '';
-		let afterPrice = sign;
-		// isPercent T  currencySignB4 T  sign appears  afterPrice
-		// isPercent T  currencySignB4 F  sign appears  afterPrice
-		// isPercent F  currencySignB4 F  sign appears  afterPrice
-		// isPercent F  currencySignB4 T  sign appears  b4Price
-		if (currencySignB4 && !price.isPercent) {
-			b4Price = sign;
-			afterPrice = '';
-		}
-		const actions = [];
-		if (price.priceType > 1 && btn === 'trash') {
-			actions.push(<Button icon={'trash'} onClick={deletePrice} minimal />);
-		}
-		if (btn === 'add') {
-			actions.push(<Button icon={'insert'} onClick={addNewPrice} minimal />);
-		}
-		return (
-			<tr>
-				<td width={'7.5%'} style={cellStyle}>
-					{price.dbId}
-				</td>
-				<td width={'15%'} style={cellStyle}>
-					<Field
-						component={'select'}
-						initialValue={price.priceType}
-						name={`[prices][${price.id}][type]`}
-						disabled={price.priceType === 1}
-						style={inputStyle}
-					>
-						{options.map((option) => (
-							<option key={option.id} value={option.id}>
-								{option.type}
-							</option>
-						))}
-					</Field>
-				</td>
-				<td width={'20%'} style={cellStyle}>
-					<Field
-						type={'text'}
-						component={'input'}
-						initialValue={price.name}
-						name={`[prices][${price.id}][name]`}
-						placeholder={'label...'}
-						style={inputStyle}
-					/>
-				</td>
-				<td width={'30%'} style={cellStyle}>
-					<Field
-						type={'text'}
-						component={'input'}
-						initialValue={price.desc}
-						name={`[prices][${price.id}][desc]`}
-						placeholder={'description...'}
-						style={inputStyle}
-					/>
-				</td>
-				<td width={'15%'} style={amountStyle}>
-					<div style={moneyStyle}>
-						<div style={b4Style}>{b4Price}</div>
-						<div style={CurrencyStyle}>
-							<Field
-								type={'number'}
-								component={'input'}
-								initialValue={price.amount}
-								name={`[prices][${price.id}][amount]`}
-								placeholder={'amount...'}
-								style={{ margin: '0 auto', textAlign: 'right', maxWidth: '105px' }}
-							/>
-						</div>
-						<div style={aftStyle}>{afterPrice}</div>
-					</div>
-				</td>
-				<td width={'7.5%'} style={actionsStyle}>
-					{actions}
-				</td>
-			</tr>
-		);
-	};
-	const renderPriceRowWithDelete = (price) => renderPriceRow(price, 'trash');
+// just temporary
+import styles from './inlineStyles';
 
+const TicketPriceCalculatorForm = ({ form, values: { ticket, prices } }) => {
+	const calcDir = !!ticket.reverseCalculate;
+	const toggleCalcDir = () => form.mutators.toggleCalcDir('ticket.reverseCalculate');
```

Comment:
This might be ok for proto, but I would suggest that we'd rather make this less nested or make another approach.
Also, we can create an issue just for that and take care later.

Reply:
good point
ya I can explicitly pass in the mutators as separate props.